src/data/role/RoleD2ApiRepository.ts

```diff
@@ -0,0 +1,39 @@
+import { Role } from "../../domain/role/Role";
+import { RoleRepository } from "../../domain/role/RoleRepository";
+import { D2Api } from "../../types/d2-api";
+
+export class RoleD2ApiRepository implements RoleRepository {
+    constructor(private api: D2Api) {}
+    async getByName(name: string): Promise<Role | undefined> {
+        const { userRoles } = await this.api.metadata
+            .get({
+                userRoles: {
+                    fields: { id: true, name: true, publicAccess: true, description: true },
+                    filter: { displayName: { eq: name } },
```

Comment:
Not sure, just to confirm, the filter we need is over the displayName (translated) or name?